src/libs/protocols/protos/agent.proto

```diff
@@ -566,3 +567,7 @@ message ResizeVolumeRequest {
 	string volume_guest_path = 1;
 	uint64 size = 2;
 }
+
+message SetPolicyRequest {
```

Comment:
I'm not familiar with how kata-runtime runs, but does this mean that the policy is delivered from the host side. Is it covered by the tee evidence?

Reply:
+1, from the referenced slides, I can clearly see the value of pre-populated policy and KBS-provided policy. Could you explain a bit about the use case and threat model of the host-provided policy? That is the one requiring a new agent proto API.

Reply:
@Xynnn007 Sorry, we ran out of time in last week’s Kata meeting before I got a chance to talk about this. However:

1. I presented that in a CoCo meeting a while ago – at offset 44:00 from https://zoom.us/rec/share/MMDPlX90B9ouSn3weOHMqtCBh0RkJvutDn-KFGKUi4pCoUoyYKVnAVZk_FEYW1KL.SaW8DXu9gFI3JZDu

2. Similar information is present in slide 20 of my slides for the Kata group (https://docs.google.com/presentation/d/1GraNlqLrP9z8ZMEF8WDdBHv0vXmQbE_d/edit#slide=id.p20)

3. A longer discussion about this is in https://github.com/confidential-containers/confidential-containers/issues/126#issuecomment-1644402591

Please let me know if you have additional questions about this.


@bergwolf , the main advantages of the Host-provided Policy are:

a. Common, generic Guest image for multiple pods and multiple K8s customers - because we don't have to change the Guest image to provide the Policy for a Particular pod.

b. Immutable Policy - because the TEE doesn't allow anyone (even the trusted components inside the TEE!) to change the expected policy hash that gets attached to the TEE. For example, when using a SEV-SNP TEE, nobody can change the value of the HOST_DATA field after the TEE started its execution.

c. Ability to verify the expected Policy hash using a generic remote attestation service - as long as that service supports verifying the value of HOST_DATA or similar TEE field.